internal/service/advancedcluster/data_source_advanced_clusters.go

```diff
@@ -292,41 +292,25 @@ func dataSourcePluralRead(ctx context.Context, d *schema.ResourceData, meta any)
 		useReplicationSpecPerShard = v.(bool)
 	}
 
-	if !useReplicationSpecPerShard {
-		list, resp, err := connV220240530.ClustersApi.ListClusters(ctx, projectID).Execute()
```

Comment:
In previous implementation results does not return clusters that are asymmetric if user_replication_spec_per_shard is disabled (this is how old API List operation currently works). I believe now we are not handling this case and instead we would have an empty replication specs attribute for asymmetric clusters.

Reply:
@AgustinBettati @oarbusi at the the end, do we need to do different ListClusters calls or it's ok to do only one with the lastest API independently of the value of user_replication_spec_per_shard?

Reply:
as you can see in the final implementation, one call to ListClusters with the latest API is enough. Important thing is to not include the asymmetric clusters if `user_replication_spec_per_shard` is disabled